### DISCUSSION TITLE
Energy dashboard

### DISCUSSION THREAD

**Discussion by AndaleR** [nathanmarlor/foxess_modbus#620] [Support] [answered]
Just trying different things and now installed Foxess-modbus-plugin new and went through the setup.

Last step was to automatically configure the energy-dashboard in Home Assistant. But: There are strange values going to this dashboard? `grid consumption total` `feed-in total` ...

Is there a bug in the last update(s)? Or am I wrong and those data should be fine? But when I look to the energy-dashboard, I would prefer the data from the actual day.

**Comment by FozzieUK** [comment]
The energy dashboard would normally take Riemann sums from power sensors for its input, they would then have a state of total_increasing and a device class of energy (kWh) - the energy dashboard uses that as it’s source to visualise statistics for hourly, daily, weekly, monthly etc.. 

The various total values provided by the integration are either energy statistics sourced directly from the inverter or they are produced from integrals so they will be the same i.e. total_increasing and a device class of energy (kWh) - they are exactly what the dashboard source requires. 👍

**Reply by AndaleR** [reply]
Makes sense. But: When I looked at the values right after installation, there was 148kWh infeed today - and that´s for sure wrong…

So I changed the entitities and it looks better now.

**Reply by FozzieUK** [reply, marked as answer]
If it is a new HA install the entities will have gone from having no history and a state of none to whatever the current value of the inverters ‘total’ register is - i’m guessing it was 148kWh for your import since the inverter was installed ?, if you had left it to continue collecting it would then have settled down to incremental energy increases as you would expect.

**Reply by AndaleR** [reply]
Ok. I will give it a try and installed it again - and let the energy-dashboard getting filled with the automatically stuff after installation. Will have a look at it the next days.

**Reply by abertram** [reply]
Hi @AndaleR,

is the issue solved? If yes, how?

Best,
Alex

**Reply by AndaleR** [reply]
Looks fine now.

**Reply by abertram** [reply]
How it was solved? Did you do anything?

**Reply by AndaleR** [reply]
Hm... I´m not sure - but I don't think I changed anything. Only updates?

My setting looks like this.
![Bildschirmfoto 2024-09-09 um 18 00 36](https://github.com/user-attachments/assets/8d252c67-7de4-48bc-b6bd-a766a6eb08c4)

**Comment by horstvogel** [comment]
Hello,
are this the correct values for the enery dashboard
The household consumption doesn't seem plausible to me

thanks!


![image](https://github.com/nathanmarlor/foxess_modbus/assets/25108737/aa959a61-5f08-4a0c-ab9b-1b0ba43f89d5)

**Reply by surfmk** [reply]
Hi AndaleR,
I have the same question... I have the same default energy dashboard which is wrong in many respects. I was expecting the devices would be automatically chosen to show instantaneous PV power / battery / grid consumption. Instead, we see Totals which I don't understand either. I have tried to replace the automatically chosen devices but the devices I expect to use are not classed as energy... therefore do not show up. There must be a YouTube video somewhere to get this configured as expected... and will post a reply if I find one.
Best, Paul

**Reply by surfmk** [reply]
![image](https://github.com/user-attachments/assets/314f0c2e-b1d4-4754-8e29-206fe659cb1f)

**Reply by surfmk** [reply]
My system has been running for 4 days and I am currently producing 7kW of PV, so 19.4kW doesn't make any sense...?
I have 20kWh battery capacity. My batteries are currently 92% full.

**Reply by surfmk** [reply]
Apologies, I re-read the post from FozzieUK and now understand the energy meter is supposed to use aggregates and mine is working.... but not really what I would like to see which is a live version of this:

![image](https://github.com/user-attachments/assets/ffd5d022-169e-4470-ae83-8c852d013a4e)

I'll see if I can make a new energy card that shows current consumption which is what I need to use in my automations (and therefore more useful to visualize).

**Reply by surfmk** [reply]
OK, done...

This is what I wanted... instantaneous powerflow which I can now use to make decisions about what to power-up / down etc...
I used a very nice work done by flixlix:

https://github.com/flixlix/power-flow-card-plus

<img width="797" height="685" alt="image" src="https://github.com/user-attachments/assets/e5162c9f-ac21-48cc-8faf-a76f520eb808" />
